src/Synthesizer/Converters/Language.hs

```diff
@@ -17,17 +17,20 @@ import Synthesizer.Structure           (Channel (..), Frequency, Sample,
                                         SynSound (..))
 
 
+-- | Convert the DSL to a 'SynSound'. This allows the user to add custom sounds to the 'SynSound' if they so wish.
 convertMusicPieceToSynthesizer :: MusicPiece -> SynSound
 convertMusicPieceToSynthesizer musicPiece = SynSound $ map convertInstrumentToChannel instruments
   where
     (MusicPiece instruments) = musicPiece
 
+-- | Converts an instrument block in the DSL to a Synthesizer channel.
 convertInstrumentToChannel :: Instrument -> Channel
 convertInstrumentToChannel instrument = Channel $ map (applyEnvelope noteStrike . convertNoteEventsToSoundEvents noteMap baseAmplitude) noteEvents
   where
     (Instrument baseFreq baseAmplitude noteStrike noteEvents) = instrument
     noteMap = generateNotes baseFreq
 
+-- | Convert a single (possibly modified by envelope) note event to a SoundEvent.
 convertNoteEventsToSoundEvents :: Map Note Frequency -> BaseAmplitude -> InstrumentEvent -> SoundEvent
```

Comment:
The comment says something about note event, while the type is InstrumentEvent

Reply:
Fixed in fcc0c66